simple_network_sim/network_of_populations.py

```diff
@@ -283,18 +284,22 @@ def doInternalProgressionAllNodes(dictOfNodeInternalStates, currentTime, disease
 
 NetworkOfPopulation = namedtuple("NetworkOfPopulation", ["progression", "states", "graph", "infectionMatrix"])
 
-def createNetworkOfPopulation(disasesProgressionFn, populationFn, graphFn):
+def createNetworkOfPopulation(disasesProgressionFn, populationFn, graphFn, ageInfectionMatrixFn):
     with open(disasesProgressionFn) as fp:
         progression = loaders.readCompartmentRatesByAge(fp)
     with open(populationFn) as fp:
         population = loaders.readPopulationAgeStructured(fp)
     graph = loaders.genGraphFromContactFile(graphFn)
 
-    # TODO: read this from a file
-    infectionMatrix = {}
-    for ageA in progression.keys():
-        for ageB in progression.keys():
-            infectionMatrix.setdefault(ageA, {})[ageB] = 0.2
+    infectionMatrix = loaders.MixingMatrix(ageInfectionMatrixFn)
+
+    agesInInfectionMatrix = set(infectionMatrix)
+    for age in infectionMatrix:
+        agesInInfectionMatrix.update(infectionMatrix[age])
+
+    assert agesInInfectionMatrix == set(progression.keys()), "infection matrix and progression ages mismatch"
+    assert agesInInfectionMatrix == {age for region in population.values() for age in region}, "infection matrix and population ages mismatch"
+    assert set(graph.nodes()) == set(population.keys()), "regions mismatch between graph and population"
```

Comment:
Would it be useful to also have a continuity check on agesInInfectionMatrix here? Check that there is no "missing" age gap?

Reply:
I think it's fine to handle that internally inside the loader function for the infection matrix. My goal with the asserts in this function is to make sure that all the datastructures are compatible. You shouldn't get a key error if you get a key from graph and try to read it from the population, for instance.